lib/charms/mysql/v0/backups.py

```diff
@@ -133,17 +133,24 @@ def _retrieve_s3_parameters(self) -> Tuple[Dict, List[str]]:
 
         # Add some sensible defaults (as expected by the code) for missing optional parameters
         s3_parameters.setdefault("endpoint", "https://s3.amazonaws.com")
-        s3_parameters.setdefault("region")
+        s3_parameters.setdefault("region", "")
         s3_parameters.setdefault("path", "")
+        s3_parameters.setdefault("s3-uri-style", "auto")
+        s3_parameters.setdefault("s3-api-version", "auto")
+
+        # Clean up extra slash symbols to avoid issues on 3rd-party storages
+        # like Ceph Object Gateway (radosgw)
+        s3_parameters["endpoint"] = s3_parameters["endpoint"].rstrip("/")
+        s3_parameters["path"] = s3_parameters["path"].strip("/")
```

Comment:
1) Should we strip spaces from both sides (if any) for all params?
2) what about to remove "/" from bucket too? to be fully protected here?

```
juju config s3-integrator \
    endpoint="https://s3.amazonaws.com/" \
    bucket="postgresql-test-bucket-1/" \
    path="/postgresql-test/ " \
    region=" us-west-2"
```

Reply:
Leta do in separate PR